### PR TITLE
Stubs for geoip2 third party library

### PR DESCRIPTION
**Pull request by jolaf**
Approved by the developer, see https://github.com/maxmind/GeoIP2-python/issues/70

### CODE REVIEW

File: third_party/3/geoip2/errors.pyi [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+from typing import Optional
+
+class GeoIP2Error(RuntimeError): ...
+
+class AddressNotFoundError(GeoIP2Error): ...
+
+class AuthenticationError(GeoIP2Error): ...
+
+class HTTPError(GeoIP2Error):
+    http_status: int = ...
```
Comment (JelleZijlstra) [inline third_party/3/geoip2/errors.pyi]: Looks like these attributes should be Optional. Also, you can remove the `= ...`.

File: third_party/3/geoip2/models.pyi [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+class SimpleModel(SimpleEquality):
+    __metaclass__: Any = ...
+
+class AnonymousIP(SimpleModel):
+    is_anonymous: bool = ...
+    is_anonymous_vpn: bool = ...
+    is_hosting_provider: bool = ...
+    is_public_proxy: bool = ...
+    is_tor_exit_node: bool = ...
+    ip_address: str = ...
```
Comment (JelleZijlstra) [inline third_party/3/geoip2/models.pyi]: Should this be Optional?

File: third_party/3/geoip2/models.pyi [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    raw: str = ...
```
Comment (JelleZijlstra) [inline third_party/3/geoip2/models.pyi]: This should be either a _RawResonse or Any.

File: third_party/3/geoip2/models.pyi [hunk 4]
```diff
@@ -0,0 +1,68 @@
+# Stubs for geoip2.models (Python 3)
+
+from typing import Any, Mapping, Optional, Sequence
+
+from geoip2 import records
+
+from geoip2.mixins import SimpleEquality
+
+_Locales = Optional[Sequence[str]]
+_RawResonse = Mapping[str, Mapping[str, Any]]
```
Comment (JelleZijlstra) [inline third_party/3/geoip2/models.pyi]: Response

File: third_party/3/geoip2/models.pyi [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+    is_anonymous_vpn: bool = ...
+    is_hosting_provider: bool = ...
+    is_public_proxy: bool = ...
+    is_tor_exit_node: bool = ...
+    ip_address: str = ...
+    raw: str = ...
+    def __init__(self, raw: str) -> None: ...
+
+class ASN(SimpleModel):
+    autonomous_system_number: int = ...
```
Comment (JelleZijlstra) [inline third_party/3/geoip2/models.pyi]: Same here and on the three classes below, the type of `raw` is wrong and it looks like the other fields are Optional (they're populated through `.get()`).

File: third_party/3/geoip2/records.pyi [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+
+class Subdivisions(Tuple[Subdivision]):
+    def __new__(cls, locales: _Locales, *subdivisions: Subdivision) -> Subdivisions: ...
+    def __init__(self, locales: _Locales, *subdivisions: Subdivision) -> None: ...
+    @property
+    def most_specific(self) -> Subdivision: ...
+
+class Traits(Record):
+    autonomous_system_number: int = ...
+    autonomous_system_organization: int = ...
```
Comment (JelleZijlstra) [inline third_party/3/geoip2/records.pyi]: unicode according to the docstring

**Comment by jolaf** [comment]
> Also, any chance these could be moved to 2and3? Probably all that is needed is to use Text instead of str.
I have no experience with stubs for Python 2, but I'll do as you say.

File: third_party/3/geoip2/errors.pyi (hunk 1 above)
Comment (jolaf) [reply third_party/3/geoip2/errors.pyi]: Yep, you're right.

File: third_party/3/geoip2/models.pyi (hunk 2 above)
Comment (jolaf) [reply third_party/3/geoip2/models.pyi]: Yep, you're right.

File: third_party/3/geoip2/models.pyi (hunk 3 above)
Comment (jolaf) [reply third_party/3/geoip2/models.pyi]: Yep, you're right.

File: third_party/3/geoip2/models.pyi (hunk 4 above)
Comment (jolaf) [reply third_party/3/geoip2/models.pyi]: Right, thanks!

File: third_party/3/geoip2/models.pyi (hunk 5 above)
Comment (jolaf) [reply third_party/3/geoip2/models.pyi]: Yep, you're right.

File: third_party/3/geoip2/records.pyi (hunk 6 above)
Comment (jolaf) [reply third_party/3/geoip2/records.pyi]: Yep, you're right.

**Comment by jolaf** [comment]
I'm closing this in favor of #3317.

**Pull request closed without merging**